src/Formatter/PlatesFormatter.php

```diff
@@ -26,27 +25,42 @@ public function __construct(Engine $engine)
      */
     public function body(PayloadInterface $payload)
     {
-        $template = $this->template($payload);
-        $template = $this->engine->make($template);
-        return $this->render($template, $payload);
+        return $this->render($payload);
     }
 
     /**
      * @param PayloadInterface $payload
+     *
      * @return string
      */
+    protected function render(PayloadInterface $payload)
+    {
+        $template = $this->template($payload);
+        $output = $this->output($payload);
+
+        return $this->engine->render($template, $output);
+    }
+
+    /**
+     * @param PayloadInterface $payload
+     *
+     * @return string Template name
+     */
     protected function template(PayloadInterface $payload)
     {
         return $payload->getOutput()['template'];
     }
 
     /**
-     * @param Template $template
      * @param PayloadInterface $payload
-     * @return string
+     *
+     * @return array $output
      */
-    protected function render(Template $template, PayloadInterface $payload)
+    protected function output(PayloadInterface $payload)
     {
-        return $template->render($payload->getOutput());
+        $output = $payload->getOutput();
+        unset($output['template']);
```

Comment:
@shadowhand In fact nothing. This code marks the problem with using PDO `Payalod::withOutput` to pass special keys for Responder services. I've read your the discussion about it [here](https://github.com/sparkphp/spark/pull/23#discussion_r36199248). 

Now such the key as `template` for a formatter:
1) _no different_ from data for a user (dual use);
2) reserved name of key in a data for a user;
3) put in a data for a user (what breaks a use case with "Content Negotiation", because unnecessary template name puts into json, xml, etc.)

I think that for debugging is better to use extensions.